starknet_devnet/blocks.py

```diff
@@ -99,85 +99,40 @@ async def generate(
 
-        if self.lite:
-            block_hash = block_number
+        if is_empty_block:
+            transaction_receipts = ()
+            transactions = []
         else:
+            signature = transaction.get_signature()
             block_hash = await calculate_block_hash(
-                general_config=state.general_config,
-                parent_hash=parent_block_hash,
-                block_number=block_number,
-                global_state_root=state_root,
-                block_timestamp=timestamp,
-                tx_hashes=[transaction.internal_tx.hash_value],
-                tx_signatures=[signature],
-                event_hashes=[],
-                sequencer_address=state.general_config.sequencer_address
-            )
-
-        transaction_receipts = (transaction.get_execution(),)
-
-        block = StarknetBlock.create(
-            block_hash=block_hash,
+            general_config=state.general_config,
+            parent_hash=parent_block_hash,
             block_number=block_number,
-            state_root=state_root,
-            transactions=[transaction.internal_tx],
-            timestamp=timestamp,
-            transaction_receipts=transaction_receipts,
-            status=BlockStatus.ACCEPTED_ON_L2,
-            gas_price=state.state.block_info.gas_price,
-            sequencer_address=state.general_config.sequencer_address,
-            parent_block_hash=parent_block_hash,
-            starknet_version=CAIRO_LANG_VERSION
-        )
-
-        self.__num2block[block_number] = block
-        self.__hash2num[block_hash] = block_number
-
-        if state_update is not None:
-            state_update = BlockStateUpdate(
-                block_hash=block_hash,
-                old_root=state_update.old_root,
-                new_root=state_update.new_root,
-                state_diff=state_update.state_diff,
+            global_state_root=state_root,
+            block_timestamp=timestamp,
+            tx_hashes=[transaction.internal_tx.hash_value],
+            tx_signatures=[signature],
+            event_hashes=[],
+            sequencer_address=state.general_config.sequencer_address
             )
-
-        self.__state_updates[block_number] = state_update
-
-        return block
-
-    def generate_empty(
-        self, state: StarknetState, state_root: bytes, state_update = None
-    ) -> StarknetBlock:
-        """
-        Generate block without transaction
-        """
-        block_number = self.get_number_of_blocks()
-        timestamp = state.state.block_info.block_timestamp
-
-        if block_number == 0:
-            parent_block_hash = 0
-        else:
-            last_block = self.__get_last_block()
-            parent_block_hash = last_block.block_hash
-
-        #Fake block number
-        block_hash = block_number
+            transaction_receipts = (transaction.get_execution(),)
+            transactions=[transaction.internal_tx]
+        if self.lite or is_empty_block:
```

Comment:
Would be good to have empty lines between adjacent if-blocks.

Reply:
> 

i ran the tests and they were not reporting anything except `too many arguments`

<img width="623" alt="Screen Shot 2022-08-26 at 9 03 06 am" src="https://user-images.githubusercontent.com/26416205/186853672-2bbce198-2d1a-4db8-b186-06e3a67f9382.png">
g



> Do you run tests before each push? `./scripts/test.sh`
> 
> I'm asking because with the last push, changes were introduced which don't just refactor but also change the behavior of the refactored code. If you ran the tests and the tests didn't report an error, that means the tests are not good.

yes, i always ran the tests. unless its green i won't push